scapy/contrib/automotive/uds.py

```diff
@@ -1006,3 +1006,60 @@ def __init__(self, sock, pkt=UDS() / UDS_TP(), interval=2):
             interval: interval between two packets
         """
         PeriodicSenderThread.__init__(self, sock, pkt, interval)
+
+
+def UDS_SessionEnumerator(sock, wait_after_reset=0.5):
+    """ Enumerates session ID's 0x0 - 0xff
+        and returns list of UDS()/UDS_DSC() packets
+        with valid session types
+
+    Args:
+        sock: socket where packets are sent
+    """
+    pkts = UDS()/UDS_DSC(diagnosticSessionType=range(0xff))
+    results = []
+    for p in pkts:
+        a = sock.sr1(p, timeout=0.5, verbose=False)
```

Comment:
You could use the argument `inter` to specify a delay between two sends. 
This line would create an generator which delivers a list of `UDS_DSC` and `UDS_ER` packets. 
```(req for tup in product(UDS()/UDS_DSC(diagnosticSessionType=range(0xff)), UDS()/UDS_ER(resetType="hardReset")) for req in tup)```